### PR TITLE
Fix todos

### PR DESCRIPTION
**Pull request by DorianGray**
Grepped the codebase for TODO... found some I thought I could tackle!

### CODE REVIEW

File: libqtile/core/manager.py [hunk 1]
```diff
@@ -399,6 +399,7 @@ def update_net_desktops(self) -> None:
         # (via current_group), and when we set up the initial groups, there
         # aren't any screens yet. This can probably be changed when #475 is
         # fixed.
+        # #475 is fixed, this is still required.
```
Comment (m-col) [inline libqtile/core/manager.py:402]: Is this missing a word?

Comment (DorianGray) [reply libqtile/core/manager.py:402]: lemme rephrase

File: libqtile/window.py [hunk 2]
```diff
@@ -195,8 +197,21 @@ def __init__(self, window, qtile):
...
+    def x(self, value):
+        _geometry_setter("x")(self, value)
+
+    @property
+    def y(self):
+        return _geometry_getter("y")(self)
+
+    @y.setter
+    def y(self, value):
+        _geometry_setter("y")(self, value)
```
Comment (m-col) [inline libqtile/window.py:214]: You prefer it this way!?

Comment (DorianGray) [reply libqtile/window.py:214]: ack, sorry, remnant of another attempt at solving this

Comment (m-col) [reply libqtile/window.py:214]: Doesn't this way require resolving the {s,g}etter every time, where before it was resolved upon class definition?

Comment (DorianGray) [reply libqtile/window.py:214]: yes

Comment (DorianGray) [reply libqtile/window.py:214]: I'm going to add a commit that gives width and height the same treatment

File: libqtile/core/manager.py [hunk 3]
```diff
@@ -399,6 +399,7 @@ def update_net_desktops(self) -> None:
         # (via current_group), and when we set up the initial groups, there
         # aren't any screens yet. This can probably be changed when #475 is
         # fixed.
+        # #475 is fixed, catching AttributeError is still necessary.
```
Comment (m-col) [inline libqtile/core/manager.py:402]: Ah I see. I misunderstood the first form lol. Thanks for updating it.

File: libqtile/window.py [hunk 4]
```diff
@@ -199,22 +199,14 @@ def __init__(self, window, qtile):
...
-    def height(self, value):
-        _geometry_setter("height")(self, value)
+    width = property(
+        fset=_geometry_setter("width"),
+        fget=_geometry_getter("width"),
+    )
+    height = property(
+        fset=_geometry_setter("height"),
+        fget=_geometry_getter("height"),
+    )
```
Comment (m-col) [inline libqtile/window.py:209]: I for one am glad to see this change too.

**Review by m-col — APPROVED** [review]
Nice, thanks for these, LGTM.

File: libqtile/core/manager.py (hunk 3 above)
Comment (DorianGray) [reply libqtile/core/manager.py:402]: Thanks for helping me get it right!

File: libqtile/window.py (hunk 4 above)
Comment (DorianGray) [reply libqtile/window.py:209]: \o/

File: libqtile/core/manager.py (hunk 1 above)
Comment (tych0) [reply libqtile/core/manager.py]: Any chance we can just delete the bits about #475 instead? I haven't evaluated whether this TODO can be fixed, but if the part about 475 is no longer relevant, let's just remove it.

**Comment by ramnes** [comment]
I just reverted dc95d81a as it broke my ROXTerm (the window wouldn't be visible and show other parts of the screen instead), so it's probably not good for other things as well.

**Comment by DorianGray** [comment]
I'll take a closer look at the icccm docs and figure out exactly what cases should get exactly what kind of configurenotify event. It must not have been as simple as just implementing 4.2.3

**Comment by Barbarossa93** [comment]
I noticed that the same thing happens to termite, another vte based terminal emulator in https://github.com/qtile/qtile/commit/99c944e3f3b0e7062ed0d6e59aff95a06bbc7d29. Perhaps vte has specific window requests? Not sure if this helps, but  I thought it might help pinpoint where the issue is

**Pull request merged**